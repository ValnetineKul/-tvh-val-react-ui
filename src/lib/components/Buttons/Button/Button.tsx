import { default as MuiButton } from '@mui/material/Button';
import type { ReactElement, MouseEventHandler, KeyboardEvent } from 'react';
import React, { forwardRef } from 'react';
import { capitalize } from '@mui/material/utils';
import type { ButtonBaseProps } from '../../ButtonBase';
import Surface from '../../Surfaces/Surface';
import type { Surface as SurfaceColor } from '../../../../themes/core';
import { useSurface } from '../../../../themes/core';
import SpinnerWithBackdrop from '../../ProgressIndicators/Spinner/SpinnerWithBackdrop';
import useStyles from './Button.styles';
import useGetButtonProps from './helpers/useGetButtonProps';
import type { DataAttributes } from '../../../types/common';

export interface ButtonCommonProps extends ButtonBaseProps {
  variant?: 'primary' | 'secondary' | 'tertiary' | 'link';
  type?: 'button' | 'submit';
  linkType?: 'default' | 'primary';
  label: string | ReactElement;
  size?: 'sm' | 'md' | 'lg';
  disabled?: boolean;
  isLoading?: boolean;
  fullWidth?: boolean;
  className?: string;
  surfaceClassName?: string;
  containerClassName?: string;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  onKeyPress?: (event: KeyboardEvent<HTMLButtonElement>) => void;
  buttonProps?: DataAttributes;
}

interface NoIcon {
  startIcon?: never;
  endIcon?: never;
}
interface StartIcon {
  startIcon: ReactElement;
  endIcon?: never;
}
interface EndIcon {
  startIcon?: never;
  endIcon: ReactElement;
}

export type ButtonIconProps = NoIcon | StartIcon | EndIcon;

export type ButtonProps = ButtonCommonProps & ButtonIconProps;

const Button = forwardRef<HTMLButtonElement | HTMLAnchorElement, ButtonProps>(
  (
    {
      variant = 'primary',
      type = 'button',
      label,
      onClick,
      size = 'md',
      fullWidth = false,
      isLoading = false,
      className,
      surfaceClassName,
      containerClassName,
      ...props
    },
    ref
  ) => {
    const { startIcon, endIcon, linkType, buttonProps, ...restProps } = props;

    const { isLink, shouldShowLoader, buttonBaseProps } = useGetButtonProps(isLoading, restProps, variant);
    const shouldShowFullWidth = fullWidth && variant !== 'link';

    const { color: onSurface } = useSurface();

    let variantColor: SurfaceColor;
    switch (variant) {
      case 'primary':
        variantColor = 'Primary';
        break;
      case 'secondary':
        variantColor = 'Secondary';
        break;
      case 'tertiary':
        variantColor = onSurface;
        break;
      case 'link':
        variantColor = onSurface;
        break;
      default:
        variantColor = 'Primary';
    }

    const { classes, cx } = useStyles({ variantColor });
    const sizeClassName = variant !== 'link' && (classes as Record<string, string>)[`size${capitalize(size)}`];

    return (
      <Surface
        color={variantColor}
        className={cx(classes.surface, { [classes.fullWidth]: shouldShowFullWidth }, surfaceClassName)}
      >
        <SpinnerWithBackdrop
          className={cx(
            { [classes.tertiaryButtonBackdrop]: variant === 'tertiary', [classes.fullWidth]: shouldShowFullWidth },
            containerClassName
          )}
          isLoading={shouldShowLoader}
          isBackdropOpaque
        >
          <MuiButton
            type={isLink ? undefined : type}
            className={cx(
              classes.root,
              sizeClassName,
              {
                [classes.button]: variant !== 'link',
                [classes.tertiaryButton]: variant === 'tertiary',
                [classes.variantLink]: variant === 'link',
                [classes.linkTypeDefault]: variant === 'link' && linkType !== 'primary',
                [classes.linkTypePrimary]: variant === 'link' && linkType === 'primary',
              },
              className
            )}
            classes={{
              startIcon: classes.startIcon,
              endIcon: classes.endIcon,
            }}
            startIcon={startIcon}
            endIcon={endIcon}
            fullWidth={fullWidth}
            onClick={onClick}
            disableElevation
            ref={ref}
            {...restProps}
            {...buttonBaseProps}
            {...buttonProps}
          >
            {label}
          </MuiButton>
        </SpinnerWithBackdrop>
      </Surface>
    );
  }
);
Button.displayName = 'Button';

export default Button;
