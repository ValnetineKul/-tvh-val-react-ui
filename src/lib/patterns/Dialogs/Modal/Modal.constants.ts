export const ARIA_CLOSE_DIALOG = 'Close dialog';
