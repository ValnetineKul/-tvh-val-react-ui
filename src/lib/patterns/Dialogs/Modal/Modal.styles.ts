import { makeStyles } from '../../../../themes/core';

const SMALL_WIDTH = 300;
const MEDIUM_WIDTH = 600;
const LARGE_WIDTH = 1224;

const useStyles = makeStyles({ name: 'Modal' })((theme) => ({
  root: {
    display: 'flex',
    justifyContent: 'space-between',
    zIndex: 1200,
  },
  title: {
    display: 'flex',
    alignItems: 'center',
    padding: `${theme.layoutSpacing['spacing/300']} ${theme.layoutSpacing['spacing/400']}`,
    [theme.breakpoints.up('sm')]: {
      padding: theme.layoutSpacing['spacing/400'],
    },
  },
  titleText: {
    marginRight: 'auto',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
  },
  backdrop: {
    position: 'fixed',
    zIndex: '-1',
  },
  paper: {
    width: '100%',
    margin: theme.layoutSpacing['spacing/000'],
    boxShadow: theme.shadow['shadow/500'],
    [theme.breakpoints.up('sm')]: {
      maxWidth: '100%',
      minHeight: 186,
      maxHeight: '90vh',
      marginTop: theme.layoutSpacing['spacing/000'],
    },
  },
  sizeSm: {
    [theme.breakpoints.up('sm')]: {
      width: SMALL_WIDTH,
    },
  },
  sizeMd: {
    [theme.breakpoints.up('md')]: {
      width: MEDIUM_WIDTH,
    },
  },
  sizeLg: {
    [theme.breakpoints.up('sm')]: {
      maxWidth: LARGE_WIDTH,
      marginLeft: theme.layoutSpacing['spacing/450'],
      marginRight: theme.layoutSpacing['spacing/450'],
    },
  },
  contentWrapper: {
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    overflowY: 'auto',
  },
}));

export default useStyles;
