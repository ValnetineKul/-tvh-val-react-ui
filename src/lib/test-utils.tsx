import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
