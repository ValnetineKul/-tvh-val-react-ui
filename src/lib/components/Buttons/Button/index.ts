import Button from './Button';
import type { ButtonProps, ButtonCommonProps, ButtonIconProps } from './Button';

export default Button;
export type { ButtonProps, ButtonCommonProps, ButtonIconProps };
