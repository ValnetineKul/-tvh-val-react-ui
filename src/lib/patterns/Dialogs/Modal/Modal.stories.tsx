import type { ComponentProps } from 'react';
import React, { useState } from 'react';
import useScreenSize from '../../../hooks/useScreenSize';
import Button from '../../../components/Buttons/Button';
import { FileUpload } from '../../../components/Icon/icons/functional';
import Icon from '../../../components/Icon';
import TextField from '../../../components/TextField';
import Typography from '../../../components/Typography';
import { createMeta } from '../../../story-utils';
import type { StoryTemplate } from '../../../story-utils';
import DialogContent from '../DialogContent';
import DialogFooter from '../DialogFooter';
import Modal from './Modal';

type Props = ComponentProps<typeof Modal>;
type Args = Omit<Props, 'open' | 'onClose'>;

const Template = ((args: Args) => {
  const [isOpen, setIsOpen] = useState(false);

  const handleClickOpen = () => {
    setIsOpen(true);
  };

  const handleClose = () => {
    setIsOpen(false);
  };

  const handleSubmit = (event: React.FormEvent) => {
    event.preventDefault();
    setIsOpen(false);
  };

  const isDesktop = useScreenSize('Tablet').isTabletUp;

  return (
    <>
      <Button label="Open modal" variant="tertiary" onClick={handleClickOpen} />
      <Modal {...args} open={isOpen} onClose={handleClose}>
        <form onSubmit={handleSubmit}>
          <DialogContent>
            <Typography variant="body500">
              Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor.
            </Typography>
            <div style={{ marginTop: '16px' }}>
              <Button startIcon={<Icon icon={FileUpload} title="start" />} variant="secondary" label="Upload file" />
            </div>
            <ul style={{ listStyle: 'none', margin: 0, padding: 0 }}>
              <li style={{ marginTop: '16px' }}>
                <TextField type="text" label="Label" placeholder="Placeholder" fullWidth required />
              </li>
              <li style={{ marginTop: '16px' }}>
                <TextField type="text" label="Label" placeholder="Placeholder" fullWidth required />
              </li>
              <li style={{ marginTop: '16px' }}>
                <TextField type="text" label="Label" placeholder="Placeholder" fullWidth required />
              </li>
              <li style={{ marginTop: '16px' }}>
                <TextField type="text" label="Label" placeholder="Placeholder" fullWidth required />
              </li>
              <li style={{ marginTop: '16px' }}>
                <TextField type="text" label="Label" placeholder="Placeholder" fullWidth required />
              </li>
            </ul>
          </DialogContent>
          <DialogFooter
            direction={isDesktop && args.size !== 'sm' ? 'horizontal' : 'vertical'}
            actionButtons={[
              <Button type="submit" label="Primary action" variant="primary" />,
              <Button label="Cancel" variant="tertiary" onClick={handleClose} />,
            ]}
            alternativeAction="AlternativeAction"
          />
        </form>
      </Modal>
    </>
  );
}) as StoryTemplate<Args>;

export const Primitive = Template.bind({});
Primitive.args = {
  size: 'md',
  id: 'dialog-title',
  title: 'Dialog header',
};

Primitive.storyName = 'Modal';

Primitive.argTypes = {
  content: {
    table: {
      disable: false,
    },
  },
};

export default createMeta({
  component: Modal,
  title: 'Patterns/Dialogs/Modal',
  argTypes: {
    onClose: {
      action: { argTypesRegex: '^on.*' },
      control: { disable: true },
      table: {
        type: {
          summary: '(event: any) => void',
        },
      },
    },
    id: {
      control: {
        disable: true,
      },
      table: {
        type: { summary: 'aria-labelledby={id} refers to the modal title with id={id}' },
      },
    },
    open: {
      control: {
        disable: true,
      },
      table: {
        type: { summary: 'If true, the modal is open' },
      },
    },
    ref: {
      control: {
        disable: true,
      },
      table: {
        type: { summary: 'Used for transition' },
      },
    },
  },
});
