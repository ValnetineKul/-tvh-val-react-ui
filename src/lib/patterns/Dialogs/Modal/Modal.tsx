import { default as MuiDialog } from '@mui/material/Dialog';
import DialogTitle from '@mui/material/DialogTitle';
import Slide from '@mui/material/Slide';

import type { ComponentProps, FC } from 'react';
import React, { cloneElement, forwardRef, isValidElement } from 'react';
import { capitalize } from '@mui/material/utils';
import type { PaperProps } from '@mui/material/Paper';
import type { BackdropProps } from '@mui/material/Backdrop';
import useScreenSize from '../../../hooks/useScreenSize';
import IconButton from '../../../components/Buttons/IconButton';
import Divider from '../../../components/Divider';
import { Times } from '../../../components/Icon/icons/functional';
import Icon from '../../../components/Icon';
import Surface from '../../../components/Surfaces/Surface';
import Typography from '../../../components/Typography';
import useStyles from './Modal.styles';
import { ARIA_CLOSE_DIALOG } from './Modal.constants';
import Scrim from '../../../components/Scrim';
import type { DataAttributes } from '../../../types/common';
import { useComponentCountController } from '../../../hooks/useComponentCounterHooks/useComponentCounterHooks';

const Transition = forwardRef(
  (props: ComponentProps<typeof Slide> & { children?: React.ReactElement }, ref: React.Ref<unknown>) => {
    const style = { width: '100%' };
    return <Slide direction="up" ref={ref} {...props} style={style} />;
  }
);

const PaperComponent: FC<PaperProps> = ({ elevation, ...paperProps }) => (
  <Surface {...paperProps} color="100">
    {paperProps.children}
  </Surface>
);

const BackdropComponent: FC<BackdropProps> = ({ children, className, open, ...scrimProps }) => {
  const { classes, cx } = useStyles();

  return (
    <Scrim
      {...scrimProps}
      className={cx(className, classes.backdrop)}
      isOpen={open}
      variant="dark"
      innerChildren={children}
    />
  );
};

export interface ModalProps {
  size?: 'sm' | 'md' | 'lg';
  id: string;
  open: boolean;
  title: string | React.ReactNode;
  onClose?: React.MouseEventHandler<HTMLButtonElement>;
  className?: string;
  rootClassName?: string;
  titleClassName?: string;
  containerClassName?: string;
  paperClassName?: string;
  TransitionProps?: {
    onEntered: () => void;
    onExiting: () => void;
  };
  closeButtonProps?: React.HTMLAttributes<HTMLButtonElement> & DataAttributes;
}

const Modal: FC<ModalProps> = ({
  size = 'md',
  id,
  open,
  title,
  onClose,
  children,
  rootClassName,
  titleClassName,
  containerClassName,
  paperClassName,
  TransitionProps,
  closeButtonProps,
  ...props
}) => {
  useComponentCountController('Modal', !!open);
  const { classes, cx } = useStyles();
  const { isTablet, isTabletUp } = useScreenSize('Tablet');

  const content = isValidElement(children)
    ? cloneElement(children, {
        className: cx(classes.contentWrapper, children.props.className),
      })
    : children;
  const actualSize = isTablet && size === 'md' ? 'lg' : size;
  const sizeClassName = (classes as Record<string, string>)[`size${capitalize(actualSize)}`];

  return (
    <MuiDialog
      open={open}
      fullScreen={!isTabletUp}
      onClose={onClose}
      aria-labelledby={id}
      TransitionComponent={Transition}
      BackdropComponent={BackdropComponent}
      classes={{
        root: cx(classes.root, rootClassName),
        paper: cx(classes.paper, { [sizeClassName]: isTabletUp }, paperClassName),
        container: cx(containerClassName),
      }}
      PaperComponent={PaperComponent}
      TransitionProps={TransitionProps}
      {...props}
    >
      <DialogTitle id={id} className={cx(classes.title, titleClassName)}>
        {typeof title === 'string' ? (
          <Typography variant="h4" className={classes.titleText} component="span">
            {title}
          </Typography>
        ) : (
          title
        )}
        {onClose && (
          <IconButton
            {...closeButtonProps}
            size="md"
            onClick={onClose}
            icon={<Icon icon={Times} />}
            aria-label={ARIA_CLOSE_DIALOG}
          />
        )}
      </DialogTitle>
      <Divider direction="horizontal" />
      {content}
    </MuiDialog>
  );
};

export default Modal;
