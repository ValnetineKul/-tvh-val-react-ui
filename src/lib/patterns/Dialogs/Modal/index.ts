import Modal from './Modal';

export { ARIA_CLOSE_DIALOG } from './Modal.constants';
export type { ModalProps } from './Modal';
export default Modal;
