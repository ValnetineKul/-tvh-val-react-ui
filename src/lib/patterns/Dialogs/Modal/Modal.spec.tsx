import React from 'react';
import { screen, userEvent, render, initResponsiveTest } from '../../../test-utils';
import Button from '../../../components/Buttons/Button';
import Modal from './Modal';

const props = {
  open: true,
  id: 'dialog-title',
  title: 'Dialog header',
};

describe('Dialog', () => {
  afterEach(() => jest.clearAllMocks());

  it('Should render correctly', () => {
    const { baseElement } = render(<Modal size="sm" {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('Should callback on header close button click', () => {
    const onClose = jest.fn();
    render(<Modal size="sm" onClose={onClose} {...props} />);
    userEvent.click(screen.getByRole('button'));
    expect(onClose).toHaveBeenCalledTimes(1);
  });

  it('Should not render header close button if there is no onClose prop (close button is optional)', () => {
    render(<Modal size="sm" {...props} />);
    expect(screen.queryByRole('button')).not.toBeInTheDocument();
  });

  describe('Should set modal body size', () => {
    it('Should set full screen modal body on mobile', () => {
      initResponsiveTest('Mobile');

      render(<Modal {...props} />);
      expect(screen.getByRole('dialog').className).toMatch('MuiDialog-paperFullScreen');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeSm');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeMd');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeLg');
    });

    it('Should set "sm" modal body on tablet', () => {
      initResponsiveTest('Tablet');

      render(<Modal size="sm" {...props} />);
      expect(screen.getByRole('dialog')?.className).toMatch('Modal-sizeSm');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeMd');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeLg');
    });

    it('Should set "sm" modal body on desktop', () => {
      initResponsiveTest('Desktop');

      render(<Modal size="sm" {...props} />);
      expect(screen.getByRole('dialog')?.className).toMatch('Modal-sizeSm');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeMd');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeLg');
    });

    it('Should not set "md" modal body on tablet', () => {
      initResponsiveTest('Tablet');

      render(<Modal size="md" {...props} />);
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeMd');
      expect(screen.getByRole('dialog')?.className).toMatch('Modal-sizeLg');
    });

    it('Should set "md" modal body on desktop', () => {
      initResponsiveTest('Desktop');

      render(<Modal size="md" {...props} />);
      expect(screen.getByRole('dialog')?.className).toMatch('Modal-sizeMd');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeSm');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeLg');
    });

    it('Should set "lg" modal body on tablet', () => {
      initResponsiveTest('Tablet');

      render(<Modal size="lg" {...props} />);
      expect(screen.getByRole('dialog')?.className).toMatch('Modal-sizeLg');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeSm');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeMd');
    });

    it('Should set "lg" modal body on desktop', () => {
      initResponsiveTest('Desktop');

      render(<Modal size="lg" {...props} />);
      expect(screen.getByRole('dialog')?.className).toMatch('Modal-sizeLg');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeSm');
      expect(screen.queryByRole('dialog')?.className).not.toMatch('Modal-sizeMd');
    });
  });
  it('Should render custom title', () => {
    const args = {
      ...props,
      title: <Button label="Custom title" />,
    };
    render(<Modal {...args} />);
    const title = screen.getByRole('button', { name: 'Custom title' });
    expect(title).toBeInTheDocument();
  });
});
