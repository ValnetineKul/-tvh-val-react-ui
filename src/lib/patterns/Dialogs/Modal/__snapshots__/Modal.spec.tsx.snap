// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Should render correctly 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root tss-1fyp359-Modal-root MuiDialog-root mui-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      class="tss-btvgcx-Scrim-root"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root tss-ghinzr-Scrim-backdropRoot-Scrim-rootDark-Modal-backdrop mui-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
    </div>
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper mui-hz1bth-MuiDialog-container"
      role="presentation"
      style="width: 100%; webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="dialog-title"
        class="MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen mui-1kfceu2-MuiDialog-paper tss-b5dlcr-Surface-root-Modal-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root tss-1u1hkf4-Modal-title mui-1pxj1do-MuiTypography-root-MuiDialogTitle-root"
          id="dialog-title"
        >
          <span
            class="tss-6-Typography-headerFunctional-ref tss-1rwf52o-Typography-root-Typography-headerFunctional-Modal-titleText-Typography-h4"
          >
            Dialog header
          </span>
        </h2>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth tss-1pam8iy-Divider-root-Divider-horizontal mui-9mgopn-MuiDivider-root"
        />
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
